TerrariaServerAPI/TerrariaApi.Server/Hooking/ItemHooks.cs

```diff
@@ -15,36 +15,34 @@ public static void AttachTo(HookManager hookManager)
 		{
 			_hookManager = hookManager;
 
-			Hooks.Item.PreSetDefaultsById = OnPreSetDefaultsById;
-			Hooks.Item.PreNetDefaults = OnPreNetDefaults;
-			Hooks.Chest.QuickStack = OnQuickStack;
+			On.Terraria.Item.SetDefaults_int_bool += OnSetDefaults;
+			On.Terraria.Item.netDefaults += OnNetDefaults;
+
+			Hooks.Chest.QuickStack += OnQuickStack;
 		}
 
-		static HookResult OnPreSetDefaultsById(Item item, ref int type, ref bool noMatCheck)
+		private static void OnNetDefaults(On.Terraria.Item.orig_netDefaults orig, Item self, int type)
```

Comment:
I think it stands to make more sense to call `Item` `item`, because `self` is a keyword in some languages to refer to `this object` and that already tripped me up once.